src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMRevision.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, Nikolas Falco
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket;
+
+import com.cloudbees.jenkins.plugins.bitbucket.JsonParser.BitbucketDateFormat;
+import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketCommit;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.text.ParseException;
+import java.util.Date;
+import jenkins.plugins.git.AbstractGitSCMSource.SCMRevisionImpl;
+import jenkins.scm.api.SCMHead;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Represents a specific revision of a bitbucket {@link SCMHead}.
+ *
+ * @author Nikolas Falco
+ * @since 2.2.13
```

Comment:
2.2.14? :)

Reply:
Yes, it's 2.2.13 because you declare to wait this PR for next release. Let me back to home and change. Please complete the review so I can procede with just one change commit.

Reply:
I said I would wait but for how long, people kept asking for it.
Now they got it 🙌 Lucky this is a quick search and replace.

Your asking me to review 2345 additions, I will miss things. I am only human 😢
I think I got all